scripts/jestPerTestSetup.ts

```diff
@@ -220,14 +220,15 @@ function startStaticServer(): Promise<string> {
 export async function notifyRebuildComplete(
   watcher: RollupWatcher
 ): Promise<RollupWatcher> {
-  let callback: (event: RollupWatcherEvent) => void
-  await new Promise<void>((resolve, reject) => {
-    callback = (event) => {
-      if (event.code === 'END') {
-        resolve()
-      }
+  let resolveFn: undefined | (() => void)
+  const callback = (event: RollupWatcherEvent): void => {
+    if (event.code === 'END') {
+      resolveFn?.()
     }
-    watcher.on('event', callback)
+  }
+  watcher.on('event', callback)
+  await new Promise<void>((resolve) => {
+    resolveFn = resolve
   })
   return watcher.removeListener('event', callback)
```

Comment:
I think `watcher.once('change', callback)` can be used and it will simplify the code very much.

Reply:
@zheeeng did you add the commit to go to `.once` and rolled it back because something failed? I see your commit but looks like you force pushed afterwards

Reply:
@zheeeng did you add the commit to go to `.once` and rolled it back because something failed? I see your commit but looks like you force pushed afterwards

Reply:
> @zheeeng did you add the commit to go to `.once` and rolled it back because something failed? I see your commit but looks like you force pushed afterwards

Yeah, the `.once` version can't pass tests martic and I rolled back to the previous version.

Tested
```ts
export async function notifyRebuildComplete(
  watcher: RollupWatcher
): Promise<RollupWatcher> {
  let resolveFn: undefined | (() => void)
  const callback = (event: RollupWatcherEvent): void => {
    if (event.code === 'END') {
      resolveFn?.()
    }
  }
  watcher.on('event', callback)
  await new Promise<void>((resolve) => {
    resolveFn = resolve
  })
  return watcher.removeListener('event', callback)
}
```

```ts
export async function notifyRebuildComplete(
  watcher: RollupWatcher
): Promise<RollupWatcher> {
  await new Promise<void>((resolve) => {
    watcher.once('event', (event) => {
      if (event.code === 'END') {
        resolve()
      }
    })
  })

  return watcher
}
```

I think the last version is the simplest but failed. I'm not sure what the internal mechanism causes this issue.

Reply:
Don't think `once` work as we have the `if (event.code === 'END')` check, so the event have to be called repeatedly.

Reply:
My bad. 😔 @bluwy's description is right. It can't be `.once`.